slave/slave.c

```diff
@@ -0,0 +1,232 @@
+/**
+ * Slave Remote Query Handling
+ */
+
+#include "../rpc/gen/slave.h"
+#include "../rpc/vote.h"
+
+#include "../master/slavelist.h"
+
+#include "../../bitmap-engine/BitmapEngine/src/seg-util/SegUtil.h"
+#include "../../bitmap-engine/BitmapEngine/src/wah/WAHQuery.h"
+
+#include <math.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <pthread.h>
+#include <unistd.h>
+
+query_result *get_vector(u_int vec_id) {
+    /* Turn vec_id into the filename "vec_id.dat" */
+    int number_size = (vec_id == 0 ? 1 : (int) (log10(vec_id) + 1));
+    int filename_size = number_size + 6; /* ".dat" */
+    char filename[16];
```

Comment:
As this is now just an array of characters, we get the following warning:
```
slave/slave.c:22:9: warning: unused variable 'filename_size'
```
Not an issue, but we could clean this up by removing lines 20–22 if we wanted.